### PR TITLE
#1472: trace redux state; show missing top level fields in page editor

### PR DESCRIPTION
**Pull request by twschiller**
Part of #1472 

Subtasks:
- [x] Put relevant traces into the Redux store to simplify use across data panel and node layout
- [x] Show a Formik error if require field is not provided

How to demo:
- Add a "Alert" brick but don't configure it. Run your extension

Questions:
- @BLoe see Data Panel changes - what should happen to isLoading and error?

Future work/Out of Scope:
- Use event signals to update the trace instead of polling
- Map input validation errors for 1) nested fields, 2) other validation types

### CODE REVIEW

File: src/devTools/editor/tabs/editTab/dataPanel/DataPanel.tsx [hunk 1]
```diff
@@ -138,11 +124,10 @@ const DataPanel: React.FC<{
 
   const { values: formState } = useFormikContext<FormState>();
 
-  const [record, isLoading, error, recalculate] = useLatestTraceRecord(
-    instanceId
-  );
-
-  useInterval(recalculate, TRACE_RELOAD_MILLIS);
+  const { record } = useSelector(makeSelectBlockTrace(instanceId));
+  // FIXME: what state should flow through to the tab?
```
Comment (twschiller) [inline src/devTools/editor/tabs/editTab/dataPanel/DataPanel.tsx:128]: @BLoe can we just drop isLoading/error? Previously they would only be:
- isLoading: the initial load of the trace from the trace database 
- error: if there as an error loading the trace from the trace database

File: src/devTools/editor/tabs/editTab/dataPanel/DataPanel.tsx [hunk 2]
```diff
@@ -139,11 +125,7 @@ const DataPanel: React.FC<{
 
   const { values: formState } = useFormikContext<FormState>();
 
-  const [record, isLoading, error, recalculate] = useLatestTraceRecord(
-    instanceId
-  );
-
-  useInterval(recalculate, TRACE_RELOAD_MILLIS);
+  const { record } = useSelector(makeSelectBlockTrace(instanceId));
```
Comment (twschiller) [inline src/devTools/editor/tabs/editTab/dataPanel/DataPanel.tsx:128]: @BLoe since we aren't using useAsyncState anymore there's no notion of an isLoading/error for the retrieval of the trace itself

Comment (BLoe) [reply src/devTools/editor/tabs/editTab/dataPanel/DataPanel.tsx:128]: Ok great, sounds good

File: src/devTools/editor/tabs/editTab/dataPanel/DataPanel.tsx (hunk 1 above)
Comment (BLoe) [reply src/devTools/editor/tabs/editTab/dataPanel/DataPanel.tsx]: Yep, makes sense 👍

File: src/devTools/editor/hooks/useRuntimeErrors.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+              // Set to touched so it shows the error message
+              setFieldTouched(formikError.fieldName);
+              setFieldError(formikError.fieldName, formikError.message);
+            }
+          }
+        }
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- only run whenever there's a new trace error
+    [traceError]
```
Comment (BLoe) [inline src/devTools/editor/hooks/useRuntimeErrors.ts:90]: Are there any edge cases we need to cover here when the block pipeline changes? If blocks are re-ordered in the pipeline or something, for example. Is there any danger that formik errors would end up applied to the wrong blockInstanceId?

**Review by BLoe — APPROVED** [review]
Just a couple minor questions, but LGTM otherwise 👍

File: src/devTools/editor/hooks/useRuntimeErrors.ts (hunk 3 above)
Comment (twschiller) [reply src/devTools/editor/hooks/useRuntimeErrors.ts:90]: Hmm, let me test. There could be a corner case if you add a brick that has the same field name

File: src/devTools/editor/hooks/useExtensionTrace.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import { selectActiveExtension } from "@/devTools/editor/slices/editorSelectors";
+
+const { setExtensionTrace } = runtimeSlice.actions;
+
+const TRACE_RELOAD_MILLIS = 350;
+
+/**
+ * Hook that refreshes the trace information in the Redux store
+ */
+function useExtensionTrace() {
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/devTools/editor/hooks/useExtensionTrace.ts:31]: Yes, we eventually need to have the devtools subscribe to event (e.g., RunComplete at the end of `reducePipeline`). It's a bit complicated because the write happens via the contentScript but needs to be seen by the devtools. I don't think IDB support native event handlers, e.g., events when a record is written to a table
